File: packages/java-sdk/src/main/java/com/turbodocx/models/CreateEmbeddedSignatureResponse.java
package com.turbodocx.models;

import java.util.List;

/**
 * Response from {@code TurboSign.createEmbeddedSignature}: the document and a per-recipient
 * embed URL, in signing order.
 */
public class CreateEmbeddedSignatureResponse {
    private final String documentId;
    private final List<EmbeddedSignatureRecipientResult> recipients;

    public CreateEmbeddedSignatureResponse(String documentId, List<EmbeddedSignatureRecipientResult> recipients) {
        this.documentId = documentId;
        this.recipients = recipients;
    }

    public String getDocumentId() {
        return documentId;
    }

    public List<EmbeddedSignatureRecipientResult> getRecipients() {
        return recipients;
    }
}

File: packages/php-sdk/src/Types/Responses/CreateEmbeddedSignatureResponse.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Responses;

/**
 * Response from {@see \TurboDocx\TurboSign::createEmbeddedSignature()}: the document plus a
 * per-recipient embed URL. Mirrors the TypeScript SDK's `CreateEmbeddedSignatureResponse`.
 */
final class CreateEmbeddedSignatureResponse
{
    /**
     * @param string $documentId The created document's id
     * @param array<EmbeddedSignatureRecipientResult> $recipients One entry per signer, in signing order
     */
    public function __construct(
        public string $documentId,
        public array $recipients,
    ) {}
}
